gxa/src/main/java/uk/ac/ebi/atlas/controllers/rest/DifferentialJsonResultsParser.java

```diff
@@ -0,0 +1,93 @@
+package uk.ac.ebi.atlas.controllers.rest;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.ebi.atlas.controllers.JsonExceptionHandlingController;
+import uk.ac.ebi.atlas.model.experiment.differential.Contrast;
+import uk.ac.ebi.atlas.model.experiment.differential.DifferentialExpression;
+import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayExpression;
+import uk.ac.ebi.atlas.search.analyticsindex.differential.DifferentialResultsReader;
+import uk.ac.ebi.atlas.search.diffanalytics.DiffAnalytics;
+import uk.ac.ebi.atlas.trader.ContrastTrader;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Optional;
+
+@Named
+public class DifferentialJsonResultsParser extends JsonExceptionHandlingController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DifferentialResultsReader.class);
+
+    private ContrastTrader contrastTrader;
+
+    @Inject
+    public DifferentialJsonResultsParser(ContrastTrader contrastTrader)
+    {
```

Comment:
We don’t have a style guide, but the project follows loosely the [Java K&R style](https://en.wikipedia.org/wiki/Indentation_style#Variant:_Java). It’s not super important but try to follow the style of the project.

Reply:
Will keep that in mind for future.